### PR TITLE
Mastering the IDE tasks

### PR DESCRIPTION
**Pull request by medvector**
https://youtrack.jetbrains.com/issue/EDC-392

### CODE REVIEW

**Comment by medvector** [comment]
Applied all the comments from @stephen-hero

File: Common Programming Concepts/Data Types/Understanding Types/task.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Mastering the IDE: Understanding Types
+
+Let's examine the code from the [Slices](course://Common Programming Concepts/Data Types/Slices) task more closely.
```
Comment (Geravant) [inline Common Programming Concepts/Data Types/Understanding Types/task.md:3]: The Slices task [was moved](https://github.com/jetbrains-academy/rustlings-course/commit/e64b1b16b2e19e2ca865550c911a05d8fd8ac850) further in the curriculum, into the Understanding Ownership - Move Semantics lesson. I suggest removing the link and not referring to a parent at this point.

File: Common Programming Concepts/Functions/Extract Method Refactoring/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+fn main() {
+    let year1993 = 1993;
+    println!("{}: 10 years ago was {}", year1993, year1993 - 10);
+
+    let year2020 = 2020;
```
Comment (Geravant) [inline Common Programming Concepts/Functions/Extract Method Refactoring/src/main.rs:5]: I understand, this is just an abstract example, and we probably will not change this over the years, but as we release it in 2021, maybe we should switch to year2021 = 2021?

File: Common Programming Concepts/Functions/Extract Method Refactoring/task.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Perform the refactoring and create a new function to replace the duplicated code.
+
+Select the first occurrence of 
+
+```rust
+println!("{}: 10 years ago was {}", year1993, year1993 - 10);
+```
+
+and then either press &shortcut:ExtractMethod; or choose *Refactor&rarr;Extract Method...* from the right-click menu.
```
Comment (Geravant) [inline Common Programming Concepts/Functions/Extract Method Refactoring/task.md:24]: Unfortunately, the \&rarr; is not rendered correctly on WIn10/ 2021.5-2021.1-637 plugin, it is rendered as a "?"
Maybe we can switch to -> for the time being? I'll create an issue about rendering in the YT

File: Common Programming Concepts/Variables/Introduce Variable Refactoring/task.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+This way, future developers coming across this code would be able to fix it by changing only one place instead of three places.
+
+Let's also refactor the code!
+
+%IDE_NAME% has a handy functionality called **"Introduce Variable Refactoring"**, which allows you to quickly do this.
+
+Select any occurrence of `1993` and then either press &shortcut:IntroduceVariable; 
+or choose *Refactor&rarr;Introduce Variable...* from the right-click menu. 
+
+*Note*: you can also use &shortcut:IntroduceConstant; or *Refactor&rarr;Introduce Constant...* from the right-click menu
```
Comment (Geravant) [inline Common Programming Concepts/Variables/Introduce Variable Refactoring/task.md:17]: Unfortunately, the &rarr; is not rendered correctly on WIn10/ 2021.5-2021.1-637 plugin, it is rendered as a "?"
Maybe we can switch to -> for the time being? I'll create an issue about rendering in the YT

**Review by Geravant — CHANGES_REQUESTED** [review]

**Comment by medvector** [comment]
Fixed all the comments by @Geravant 
Please resolve comments.

File: Common Programming Concepts/If/Managing Conditions/task.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+## Mastering the IDE: Managing Conditions
+
+Can you quickly understand what the code from `main.rs` file prints?
```
Comment (stephen-hero) [inline Common Programming Concepts/If/Managing Conditions/task.md:3]: Can you quickly understand what the code from the `main.rs` file prints?

File: Common Programming Concepts/If/Managing Conditions/task.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ 
+%IDE_NAME% has lots of useful features that can help you make the code more readable.
+
+For example, let's apply [DeMorgan's law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) **quick-fix**.
+
+Place the cursor on `&&` and use the &shortcut:ShowIntentionActions; shortcut to see available context actions.
+
+![](image.png)
+
+Now, can you quickly understand what the code from `main.rs` file prints?
```
Comment (stephen-hero) [inline Common Programming Concepts/If/Managing Conditions/task.md:13]: Now, can you quickly understand what the code from the `main.rs` file prints?

**Comment by medvector** [comment]
Applied the latest comments.

**Review by Geravant — APPROVED** [review]

**Pull request merged**